Source/model/draw/Align.swift

```diff
@@ -1,5 +1,36 @@
-public enum Align {
-    case min
-    case mid
-    case max
+open class Align {
+    
+    public static let min: Align = MinAlign()
+    public static let mid: Align = MidAlign()
+    public static let max: Align = MaxAlign()
+    
+    open func align(x: Double, y: Double) -> Double {
+        return 0
+    }
+    
+    open func align(x: Double) -> Double {
+        return align(x: x, y: 0)
+    }
+    
+}
+
+private class MinAlign : Align {
+    
+    override func align(x: Double, y: Double) -> Double {
+        return 0
+    }
+}
+
+private class MidAlign : Align {
+    
+    override func align(x: Double, y: Double) -> Double {
+        return x / 2 - y / 2
```

Comment:
`(x - y) / 2` will give you 2 operations instead of 3 😉. Not a big deal, but it's a good thing to keep performance in mind.